### PR TITLE
Feature/building configurations

### PR DESCRIPTION
**Pull request by emilianowearemobile**
[FARM-206]

#### 📲 What

Add building configurations and flavors.

#### 🤔 Why
		
Development config was required by devs.
Release config was required by CI.
This is the method to have both of them.
		
#### ✅ Acceptance criteria

- [x] Manually tested
- [x] Passing all tests
- [x] Rebased/merged with latest changes from dev/release and re-tested
- [x] Removed TODOs and FIXMEs

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -31,6 +31,12 @@ if (keystorePropertiesFile.exists()) {
     keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
 }
 
+
+//
+// Project won't compile for android if you don't have google-services.json and key.properties.
```
Comment (maribel4amido) [inline android/app/build.gradle:35]: Please specify the location required for each file.

File: android/app/build.gradle [hunk 2]
```diff
@@ -51,33 +57,54 @@ android {
...
     buildTypes {
-        release {
-            // To release we need to switch to signingConfigs.release
+        debug {
+            // default
             signingConfig signingConfigs.debug
+        }
+
+        release {
+            // add --release to run command
```
Comment (maribel4amido) [inline android/app/build.gradle:74]: Which command is this? flutter or gradle?

File: android/app/build.gradle [hunk 3]
```diff
@@ -51,33 +57,54 @@ android {
...
+    flavorDimensions "flavors"
+
+    productFlavors {
+        // Add particular setup into different app flavors.
+        // Also, you can create a different main to define global variables.
+        // Recommended read https://medium.com/flutter-community/flutter-ready-to-go-e59873f9d7de
+        development {
+            // To run  flutter run --flavor development -v lib/main.dart
+            dimension "flavors"
+            applicationId = "co.farmsmart.app.dev"
```
Comment (maribel4amido) [inline android/app/build.gradle:93]: Does this mean that debug and release apks can be installed in android at the same time without overriding each other?

**Review by maribel4amido — COMMENTED** [review]
Minor improvements to the comments 
and some general questions.

File: android/app/build.gradle (hunk 1 above)
Comment (emilianowearemobile) [reply android/app/build.gradle:35]: Good idea!

File: android/app/build.gradle (hunk 2 above)
Comment (emilianowearemobile) [reply android/app/build.gradle:74]: Flutter, through terminal or IDE.

File: android/app/build.gradle (hunk 3 above)
Comment (emilianowearemobile) [reply android/app/build.gradle:93]: Exactly!

**Review by maribel4amido — APPROVED** [review]

**Pull request merged**